components/automate-cli/cmd/chef-automate/cert.go

```diff
@@ -0,0 +1,134 @@
+
+func init() {
+
+	certCmd := certCmd()
+
+	certCmd.PersistentFlags().StringVarP(
+		&certCmdFlags.hostname,
+		"hostname",
+		"n",
+		"",
+		"Hostname for the automate TLS certificate",
+	)
+	certCmd.PersistentFlags().StringVarP(
+		&certCmdFlags.filepath,
+		"file",
+		"f",
+		"",
+		"File path to save automate TLS certifcate to.",
+	)
+	RootCmd.AddCommand(certCmd)
+}
+
+func certCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "external-cert COMMAND",
+		Short: "Manage Chef Automate's external certificate",
+		Long:  "Manage Chef Automate's external certificate authority. Used for establishing TLS/SSL communication with automate.",
+	}
+
+	certShow := &cobra.Command{
+		Use:   "show",
+		Short: "Show the external TLS/SSL certificates in Automate. Optionally, save the certificates to a file in the specified path.",
+		RunE:  runCertShowCmd,
+		Args:  cobra.MaximumNArgs(2),
+	}
+
+	cmd.AddCommand(certShow)
+
+	return cmd
+}
+
+func runCertShowCmd(*cobra.Command, []string) error {
+	res, err := client.GetAutomateConfig(int64(client.DefaultClientTimeout))
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.DeploymentServiceUnreachableError,
+			"Connecting to deployment-service failed",
+		)
+	}
+	tlsCreds := res.Config.GetGlobal().GetV1().GetFrontendTls()
+
+	if certCmdFlags.hostname != "" && certCmdFlags.filepath != "" {
+		for _, tlsCred := range tlsCreds {
+			if tlsCred.GetServerName() == certCmdFlags.hostname {
+				return writeToFile(tlsCred.GetCert())
+			}
+		}
+		// Error no matching hostname found in the automate frontend_tls cert configuration
+		return status.Wrap(
+			err,
+			status.CommandExecutionError,
+			fmt.Sprintf("Unable to find matching certificate configuration for hostname %s", certCmdFlags.filepath),
+		)
+	} else if certCmdFlags.hostname != "" {
+		found := false
+		for _, tlsCred := range tlsCreds {
+			if tlsCred.GetServerName() == certCmdFlags.hostname {
+				found = true
+				writer.Printf("Hostname: %s\nCert: %s\n", tlsCred.GetServerName(), tlsCred.GetCert())
+				return nil
+			}
+		}
+		if !found {
+			// Error no matching hostname found in the automate frontend_tls cert configuration
+			return status.Wrap(
+				err,
+				status.CommandExecutionError,
+				fmt.Sprintf("Unable to find matching certificate configuration for hostname %s", certCmdFlags.filepath),
+			)
+		}
+	} else if certCmdFlags.filepath != "" {
+		if len(tlsCreds) > 1 {
+			// There is more than one cert and it is unclear which one to save to the cert file
+			return status.Wrap(
+				err,
+				status.CommandExecutionError,
+				fmt.Sprintf("Found more than one host certificate configured, please specify a hostname using the -hostname flag"),
+			)
+		}
+		return writeToFile(tlsCreds[0].GetCert())
+
+	} // Print all certs to stdout
+	for _, tlsCred := range tlsCreds {
+		if tlsCred.GetServerName() != "" {
+			writer.Printf("Hostname: %s\nCertificate:\n %s\n", tlsCred.GetServerName(), tlsCred.GetCert())
```

Comment:
Printing these headers might make it hard for users to combine this command with other commands such as `openssl`.  you might want to do `chef-automate external-cert show | openssl x509 -text` or the like

Reply:
You'd definitely want the header if you're using this to get the cert out for the use case of configuring hab supervisors to talk to automate, but printing it on stderr instead would be a good way to solve for both things.

Reply:
That's a good point! In the case of multiple hostnames would we still want to do that though, I figured the hostname would serve as an identifier and divider. Certainly for one hostname it makes more sense to forego it.

Reply:
If you print to stderr instead it would look the same in the terminal for copy/paste but would not be captured by pipe or file redirectors in the shell. Folks could do `2>&1` if they need all the output to go to a pipe or `>` for whatever reason.